src/engine/QueryPlanner.cpp

```diff
@@ -790,245 +840,476 @@ vector<QueryPlanner::SubtreePlan> QueryPlanner::seedWithScansAndText(
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromTransitiveMin(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  ParsedQuery::GraphPattern childPlan = seedFromPropertyPath(left, path._children[0], right);
+  ParsedQuery::GraphPattern p;
+  ParsedQuery::GraphPatternOperation* op = new ParsedQuery::GraphPatternOperation(ParsedQuery::GraphPatternOperation::Type::TRANS_PATH);
+  op->_pathData._left = left;
+  op->_pathData._right = right;
+  op->_pathData._min = path._limit;
+  op->_pathData._childGraphPattern = new ParsedQuery::GraphPattern(std::move(childPlan));
+  p._children.push_back(op);
+  return p;
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromTransitiveMax(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  ParsedQuery::GraphPattern childPlan = seedFromPropertyPath(left, path._children[0], right);
+  ParsedQuery::GraphPattern p;
+  ParsedQuery::GraphPatternOperation* op = new ParsedQuery::GraphPatternOperation(ParsedQuery::GraphPatternOperation::Type::TRANS_PATH);
+  op->_pathData._left = left;
+  op->_pathData._right = right;
+  op->_pathData._max = path._limit;
+  op->_pathData._childGraphPattern = new ParsedQuery::GraphPattern(std::move(childPlan));
+  p._children.push_back(op);
+  return p;
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromInverse(
+    const std::string& left, const PropertyPath& path,
+    const std::string& right) {
+  return seedFromPropertyPath(right, path, left);
+}
+
+// _____________________________________________________________________________
+ParsedQuery::GraphPattern QueryPlanner::seedFromIri(const std::string& left,
+                                                    const PropertyPath& path,
+                                                    const std::string& right) {
+  ParsedQuery::GraphPattern p;
+  p._whereClauseTriples.push_back(SparqlTriple(left, path, right));
+  return p;
+}
+
+// _____________________________________________________________________________
+std::string QueryPlanner::generateRandomVarName() {
+  constexpr int VAR_NAME_LENGTH = 32;
+  std::string name(VAR_NAME_LENGTH, ' ');
+  // pick random characters in the range [65-90] and [97-122]
+
+  // How many random characters we can extract from a single call to rand
+  int rand_bytes = (unsigned int)(std::log(RAND_MAX) / std::log(52));
+  int current_rand;
+  int bytes_left = 0;
+  for (int i = 0; i < VAR_NAME_LENGTH; i++) {
+    if (bytes_left <= 0) {
+      current_rand = rand_r(&_randomSeed);
+      bytes_left = rand_bytes;
+    }
+    uint8_t v = current_rand % 52;
+    if (v >= 26) {
+      // Characters a-z
+      v += 'a' - 26;
+    } else {
+      // Characters A-Z
+      v += 'A';
```

Comment:
Somehow the names in my test don't look very random e.g. `?impOiAAAAAAAAAAAAAAAAAAAAAAAAAAA2` and `?JHzegEAAAAAAAAAAAAAAAAAAAAAAAAAA1`